var/lib/delphix-platform/os-migration

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+MIGRATE_CONFIG_SCRIPT="/opt/delphix/migration/migrate_config.py"
+MIGRATE_CONFIG_LOG="/var/delphix/migration/log"
+
+function die() {
+	echo "$(basename "$0"): $*" >&2
+	exit 1
+}
+
+function perform_migration() {
+	zpool import -f domain0 || die "Failed to import domain0"
+	# domain0/mds is owned by delphix:staff on Illumos, and on Linux
+	# postgres can't access this dataset anymore. We change the permissions
+	# to the Linux defaults.
+	chown postgres:postgres /mds ||
```

Comment:
Should this be recursive?

Also, should this occur in the app-gate script, since the concept of MDS and the "/mds" directory is an app-gate concept? e.g. if we're migrating a system that doesn't use the virtualization application, do we still want to do this?

It's not clear to me how we split the responsibility of migration between the platform package, and the virtualization package.

Reply:
We only need to change `/mds` because all the other files are already owned by `postgres` (albeit, with group `staff` instead of group `postgres`).

I was thinking of doing this recursive chgrp asynchronously later, after the app stack has restarted, since it takes extra time and shouldn't affect the stack in any way (as only postgres accesses those files).

I have a task open to investigate the chown changes (https://jira.delphix.com/browse/LX-1775). For now it's marked as a post-stack-startup step, but if we decide that it's better to do it before starting the stack, we can always do that.